database/redis/reply/search_result.go

```diff
@@ -0,0 +1,97 @@
+package reply
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/database"
+)
+
+type searchHighlightStorageElement struct {
+	Field string `json:"field"`
+	Value string `json:"value"`
+}
+
+type searchResultStorageElement struct {
+	ObjectID   string                          `json:"object_id"`
+	Highlights []searchHighlightStorageElement `json:"highlights"`
+}
+
+func toSearchResultStorageElement(searchResult moira.SearchResult) searchResultStorageElement {
```

Comment:
I'm not sure, but maybe these functions are better written as methods of `searchResultStorageElement`?

`searchResultStorageElement.fromSearchResult` and `searchResultStorageElement.toSearchResult`?

Reply:
I am doing this in a way this done in trigger.go do save consistency between a function signatures in this package.

https://github.com/moira-alert/moira/blob/a56078419152c5a9b85c2f0f7051442345ff1062/database/redis/reply/trigger.go#L33-L52

https://github.com/moira-alert/moira/blob/a56078419152c5a9b85c2f0f7051442345ff1062/database/redis/reply/trigger.go#L54-L73